test/recipes/80-test_cmp_http_data/test_connection.csv

```diff
@@ -3,7 +3,6 @@ expected,description, -section,val, -server,val, -proxy,val, -no_proxy,val, -tls
 ,,,,,,,,,,,,,,,,,,,
 1,default config, -section,,,,,,,,BLANK,,,,BLANK,,BLANK,,BLANK,
 1,server domain name, -section,, -server,localhost:_SERVER_PORT,,,,,,,,,,,,,,
-1,server IPv6 address, -section,, -server,[::1]:_SERVER_PORT,,,,,,,,,,,,,,
```

Comment:
This should not be necessary: https://github.com/openssl/openssl/blob/01425af13f14f46acc57d0d5eebf365c5ed7c54e/test/recipes/80-test_cmp_http.t#L280

It tests that addressing by IPv6 address works if IPv6 is available. Or does it still fail with this line kept?

Reply:
1. It's not a necessary test here.
2. With the change `[::]` => `*` in apps/lib/http_server.c, the Mock server may start to accept on IPv4 (respond with "ACCEPT 0.0.0.0:nnnnn"), even if the machine supports IPv6.  So what happens is that this line *isn't* removed, and that leads to failure.

Reply:
Note that things could be designed better.  The function in apps/lib/http_server.c could be amended to take a host as well, or have a complete `-accept` string just the same as `s_server`, and then we could made better calls to test connections for all IP versions.  But done like this, with a *broken assumption* on internals?  No!

Reply:
OK